src/test/groovy/org/osiam/test/integration/RegistrationIT.groovy

```diff
@@ -348,6 +382,12 @@ class RegistrationIT extends AbstractIT {
         registeredUser.displayName == 'displayName'
 
         responseStatus == 200
+
+        //Waiting at least 5 seconds to ensure that an email was send
+        Thread.sleep(5000)
```

Comment:
you could actively poll for new emails. No need for a latch, if you don't poll in a background thread.